rollup/sync_service/sync_service.go

```diff
@@ -50,6 +51,8 @@ type SyncService struct {
 	pollInterval         time.Duration
 	latestProcessedBlock uint64
 	scope                event.SubscriptionScope
+
+	stateMu sync.Mutex // protects the service state, e.g. db and latestProcessedBlock updates
```

Comment:
I would expect simply making `latestProcessedBlock` an `atomic.Uint64` would work just fine instead of adding a mutex here.

Reply:
changed in [75413e6](https://github.com/scroll-tech/go-ethereum/pull/1044/commits/75413e676b59f6f528436ab3282dbefbf2d47e76).

Reply:
This will lead to issues when we read it in `fetchMessages`, then update in `ResetStartSyncHeight`, then overwrite the update in `fetchMessages`.

Reply:
makes sense. both "compare and swap" and "function level mutex lock" can resolve this. which code style do you prefer? @omerfirmak @Thegaram

Reply:
I think mutex is much simpler, and we don't need to worry about lock contention here. But I'm ine with either.

Reply:
after some rethinking, will change back to mutex lock for the following reasons:
(i) It's an easier change.
(ii) It's also good feedback when using the admin API in geth console: it will take action immediately after gaining the lock and finish changing the variable.

Reply:
changed back to use mutex in [a178e9e](https://github.com/scroll-tech/go-ethereum/pull/1044/commits/a178e9e8a693021dc9b72dbc450fa77e0e951137).